browser/metric_event_mapping.go

```diff
@@ -0,0 +1,43 @@
+package browser
+
+import (
+	"fmt"
+
+	"github.com/grafana/xk6-browser/common"
+)
+
+// mapMetricEvent to the JS module.
+func mapMetricEvent(vu moduleVU, cm *common.MetricEvent) (mapping, error) {
+	rt := vu.VU.Runtime()
+
+	// We're setting up the function in the Sobek context that will be reused
+	// for this VU.
+	_, err := rt.RunString(`
+	function _k6BrowserURLGroupingTest(pattern, url) {
+		let r = pattern;
+		if (typeof pattern === 'string') {
+			r = new RegExp(pattern);
+		}
+		return r.test(url);
+	}`)
+	if err != nil {
+		return nil, fmt.Errorf("evaluating regex function: %w", err)
+	}
+
+	return mapping{
+		"Tag": func(urls common.URLOverrides) error {
+			callback := func(pattern, url string) (bool, error) {
```

Comment:
I'd call this `matchRegex` or something more descriptive to be clearer and easier to understand.

Reply:
Done in 8fd880a9c1141a6a467cd1fd0dd0782b6f99deae